### ISSUE TITLE
Service type ExternalName getting 503

### ISSUE THREAD

**Issue by ulyssescb** [Kong/kong#7918]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.5.0

### Current Behavior

Some requests issued at ingresses which its backend service with `type: ExternalName` are getting `503` in about ~15% of total requests. 

Kubernetes Version: 1.17.0
Kong ingress Controller: 1.1.1

### Expected Behavior

All requests must return `200` or `404`.

### Steps To Reproduce

```yaml
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: add-authority
  labels:
    app: my-ingress
plugin: request-transformer
config:
  add:
    headers:
    -  'authority: bucket.s3.amazonaws.com'
---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  annotations:
    external-dns.alpha.kubernetes.io/ignore: "true"
    kubernetes.io/ingress.class: kong
    konghq.com/override: my-ingress
    konghq.com/plugins: add-authority
  labels:
    app: my-ingress
  name: my-ingress
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: my-service
          servicePort: 80
        path: /endpoint(/.*)?$
    host: domain.com.br
    
---
apiVersion: v1
kind: Service
metadata:
  name: my-service
  labels:
    app: my-service
  annotations:
    konghq.com/override: my-service
spec:
  type: ExternalName
  externalName: bucket.s3.amazonaws.com
  ports:
  - protocol: TCP
    port: 80
```

### Anything else?

```yaml

kong-ingress-7c5cfcc6b5-swjq2 kong-proxy 2021/09/29 20:09:00 [error] 23#0: *433506 [lua] prometheus.lua:671: log_error(): Error while setting 'upstream_target_health{upstream="my-service.namespace.80.svc",target="bucket.s3.amazonaws.com:80",address="52.217.140.137:80",state="unhealthy"}' to '0': 'no memory', client: 100.118.150.57, server: ko
ng_prometheus_exporter, request: "GET /metrics HTTP/1.1", host: "100.102.15.153:9542"

```

```yaml
kong-ingress-7c5cfcc6b5-swjq2 kong-proxy 2021/09/29 20:09:00 [error] 23#0: *433506 [lua] prometheus.lua:671: log_error(): Error while setting 'upstream_target_health{upstream="my-service.namespace.80.svc",target="bucket.s3.amazonaws.com:80",address="52.217.140.137:80",state="dns_error"}' to '0': 'no memory', client: 100.118.150.57, server: ko
ng_prometheus_exporter, request: "GET /metrics HTTP/1.1", host: "100.102.15.153:9542"
```

**Comment by hbagdi** [comment]
cc @Kong/team-k8s

**Comment by saravanan-periyasamy03** [comment]
Hi @ulyssescb , This issue is fixed in the kong 2.6 version and this GitHub issue is related to [8056](https://github.com/Kong/kong/issues/8056). Please upgrade and validate.

**Comment by mayocream** [comment]
Feel free to re-open this issue if you want to append more information.
